field/track.go

```diff
@@ -0,0 +1,404 @@
+		if len(value) == 0 || value == "=" {
+			continue
+		}
+
+		switch index {
+		case 1: // Payment card number (PAN)
+			f.PrimaryAccountNumber = value
+		case 2: // Expiration Date (ED)
+			expiredTime, timeErr := time.Parse(expiryDateFormat, value)
+			if timeErr == nil {
+				f.ExpirationDate = &expiredTime
+			}
+		case 3: // Service Code (SC)
+			f.ServiceCode = value
+		case 4: // Discretionary data (DD)
+			f.DiscretionaryData = value
+		}
+	}
+
+	return nil
+}
+
+func (f *Track) parseForVersionThird(b []byte) error {
+	matches := trackThirdRegex.FindStringSubmatch(string(b))
+	for index, val := range matches {
+		value := strings.TrimSpace(val)
+		if len(value) == 0 || value == "=" {
+			continue
+		}
+
+		switch index {
+		case 1: // Payment card number (PAN)
+			f.FormatCode = value
+		case 2: // Payment card number (PAN)
+			f.PrimaryAccountNumber = value
+		case 3: // Security Data + Additional Data
+			f.DiscretionaryData = value
+		}
+	}
+
+	return nil
+}
+
+func (f *Track) serialize() ([]byte, error) {
+
+	var pattern string
+	var handler func(index int, name string, group *syntax.Regexp, generator utils.Generator, args *utils.GeneratorArgs) string
+
+	switch f.Version {
+	case VersionFirst:
+		pattern = trackFirstPattern
+		handler = func(index int, name string, group *syntax.Regexp, generator utils.Generator, args *utils.GeneratorArgs) string {
+			var raw string
+			switch index {
+			case 0:
+				raw = f.FormatCode
+			case 1:
+				raw = f.PrimaryAccountNumber
+			case 2:
+				if len(f.Name) > 0 && f.FixedLength {
+					raw = fmt.Sprintf("%-26.26s", f.Name)
+				} else {
+					raw = f.Name
+				}
+			case 3:
+				if f.ExpirationDate != nil {
+					raw = f.ExpirationDate.Format(expiryDateFormat)
+				}
+			case 4:
+				raw = f.ServiceCode
+			case 5:
+				raw = f.DiscretionaryData
+			}
+
+			if len(raw) == 0 {
+				return `^`
+			}
+			return raw
+		}
+	case VersionSecond:
+		pattern = trackSecondPattern
+		handler = func(index int, name string, group *syntax.Regexp, generator utils.Generator, args *utils.GeneratorArgs) string {
+			var raw string
+			switch index {
+			case 0:
+				raw = f.PrimaryAccountNumber
+			case 1:
+				if f.ExpirationDate != nil {
+					raw = f.ExpirationDate.Format(expiryDateFormat)
+				}
+			case 2:
+				raw = f.ServiceCode
+			case 3:
+				raw = f.DiscretionaryData
+			}
+
+			if len(raw) == 0 {
+				return `=`
+			}
+			return raw
+		}
+	case VersionThird:
+		pattern = trackThirdPattern
+		handler = func(index int, name string, group *syntax.Regexp, generator utils.Generator, args *utils.GeneratorArgs) string {
+			var raw string
+			switch index {
+			case 0:
+				raw = f.FormatCode
+			case 1:
+				raw = f.PrimaryAccountNumber
+			case 2:
+				raw = f.DiscretionaryData
+			}
+
+			if len(raw) == 0 {
+				return `=`
+			}
+			return raw
+		}
+	}
+
+	generator, _ := utils.NewGenerator(pattern, &utils.GeneratorArgs{
+		Flags:               syntax.Perl,
+		CaptureGroupHandler: handler,
+	})
```

Comment:
So I'm understanding `regen` to be a way of generating a random string using a regular expression,
and then using these handlers to inject strings we actually want at each index. This feels long winded.

Would something like `fmt.Sprintf("%s%s%s", a, b, c)` not suffice?